selfdrive/modeld/trafficd.cc

```diff
@@ -0,0 +1,184 @@
+#pragma clang diagnostic ignored "-Wexceptions"
+#pragma clang diagnostic ignored "-Wunused"
+#include <array>
+#include <stdio.h>
+#include <stdlib.h>
+#include <signal.h>
+#include <unistd.h>
+
+#include "visionbuf.h"
+#include "visionipc_client.h"
+#include "common/timing.h"
+#include "common/clutil.h"
+#include "common/util.h"
+
+#include "models/commonmodel.h"
+#include "runners/run.h"
+#include "messaging.hpp"
+#include "thneed/thneed.h"
+
+//#include <sched.h>
+
+using namespace std;
+
+volatile sig_atomic_t do_exit = 0;
+
+const std::vector<std::string> modelLabels = {"SLOW", "GREEN", "NONE"};
+const int numLabels = modelLabels.size();
+const double modelRate = 1 / 5.;  // 5 Hz
+const bool debug_mode = true;
+
+const int original_shape[3] = {874, 1164, 3};   // global constants
+//const int original_size = 874 * 1164 * 3;
+//const int cropped_shape[3] = {665, 814, 3};
+const int cropped_size = 665 * 814 * 3;
+
+const int horizontal_crop = 175;
+const int top_crop = 0;
+const int hood_crop = 209;
+const double msToSec = 1 / 1000.;  // multiply
+const double secToMs = 1000.;
+
+
+////void sendPrediction(float *output, PubMaster &pm) {
```

Comment:
@arne182 once you have cereal working we can uncomment this and add traffic manager to manager

Reply:
wym by cereal working?